doc/developer-notes.md

```diff
@@ -331,6 +331,31 @@ Strings and formatting
 
   - *Rationale*: Bitcoin Core uses tinyformat, which is type safe. Leave them out to avoid confusion
 
+Variable names
+--------------
+
+The shadowing warning (`-Wshadow`) is enabled by default. Please name
```

Comment:
Please mention the rationale why it is enabled, why we went through this in the first place: which is to avoid bugs like #8102